### PR TITLE
Implement default and named config lookup for the kafka connector

### PR DESCRIPTION
**Pull request by cescoffier**
Implement default and named configuration lookup for the Kafka connector.

This allows the connector to resolve the configuration from named `Map` (exposed using CDI).﻿

### CODE REVIEW

**Comment by cescoffier** [comment]
@geoand this is related to service binding with multiple brokers.

**Comment by geoand** [comment]
Cool!

**Comment by Ladicek** [comment]
I wanted to file an issue eventually, but it seems I should do it sooner rather than later. We should not use `@Named`, it doesn't work as all other normal qualifiers do. (I'm sure @mkouba could swamp you with details :-) )

I'll file an issue and add the link here, before we spread `@Named` too much.

**Comment by Ladicek** [comment]
Filed https://github.com/smallrye/smallrye-reactive-messaging/issues/1096

**Comment by cescoffier** [comment]
Good point. In that case, we would need another qualifier. Something we could add to smallrye-annotations?

**Comment by Ladicek** [comment]
Would be happy to review this when rebased and using `@Identifier`.

**Comment by cescoffier** [comment]
Ah yes, good point, this need to be revisited.

File: documentation/src/main/doc/modules/connectors/partials/META-INF/connector/smallrye-kafka-incoming.adoc [hunk 1]
```diff
@@ -37,6 +37,10 @@ Type: _boolean_ | false | `true`
 
 Type: _boolean_ | false | `true`
 
+| *kafka-configuration-name* | The name set in `javax.inject.Named` of a bean that provides the default Kafka consumer/producer (`Map<String, Object>`) configuration to use for this channel. The channel configuration can still override any attribute.
```
Comment (wtrocki) [inline documentation/src/main/doc/modules/connectors/partials/META-INF/connector/smallrye-kafka-incoming.adoc:40]: So we can use for this name of the service binding that will somehow work or actually do provide extra field that will be uniquely naming the service:

I have proposed something to the spec (but we can do it ourselves)
https://github.com/k8s-service-bindings/spec/issues/187

We can do it for kafka and any future services.

[Note] this is just though - binding name is ok for now

**Comment by cescoffier** [comment]
I've recreated the PR:

1. From the main branch
2. Using @Identifier 
3. Cleaning up how we deal with the config in the connector

**Comment by Ladicek** [comment]
Looks fine otherwise. I briefly considered whether we want to add support for `@Named` for these additional config `Map`s, but probably no.

**Comment by cescoffier** [comment]
Unlike the default lookup (which still uses @Named to not break), this feature should use @Identifier. The pending work is to generate the Maps in Quarkus - which uses @Identifier already.

I was wondering if we wanted to remove the @Named support for the default strategy. I kept it, for now, I need to check with the consumers if they use that or not.

**Comment by Ladicek** [comment]
We should remove support for `@Named` one day, but I agree today is not that day.

**Comment by ozangunalp** [comment]
Can we avoid this `@Named` / `@Identifier` issue altogether by removing `-name` suffix off the config property? Same applies for the rebalance listener.

**Comment by cescoffier** [comment]
@Ladicek I've applied your suggestion (was a left-over of the very old PR)
@ozangunalp I've removed the `-name` suffix.

File: documentation/src/main/doc/modules/kafka/pages/default-configuration.adoc [hunk 2]
```diff
@@ -37,3 +37,24 @@ This previous example would extract all the configuration keys from MicroProfile
 ====
 Starting with Quarkus 1.5, a map corresponding to the previous example is automatically provided.
 ====
+
+In addition to this default configuration, you can configure the name of the `Map` producer using the `kafka-configuration-name` attribute:
```
Comment (Ladicek) [inline documentation/src/main/doc/modules/kafka/pages/default-configuration.adoc:41]: This part of documentation still uses the `-name` suffix :-)

**Comment by ozangunalp** [comment]
@cescoffier this should be rebased/reviewed after merging #1406.

**Comment by cescoffier** [comment]
If I’m not mistaken, there are 3 Pr updating the same files. I need to see
the merge order to simplify the rebase.

**Comment by cescoffier** [comment]
I removed all the remaining reference to kafka-configuration-name.

**Review by ozangunalp — APPROVED** [review]

**Pull request merged**